microservices/payment/src/services/payment-gateway/stripe.ts

```diff
@@ -247,31 +273,100 @@ class Stripe extends Abstract {
-      .object as ICheckoutEvent;
-
-    const customerEntity = await this.customerRepository.findOne({ customerId: customer });
-
-    if (!customerEntity) {
-      Log.error(`Could not find any existing customer with such customerId: ${customer}`);
+    const { id, payment_status, status } = event.data.object as ICheckoutEvent;
 
-      return;
-    }
-
-    return this.createTransaction(
+    await this.transactionRepository.update(
+      { transactionId: id },
       {
-        amount: amount_total,
-        userId: customerEntity?.userId,
         status: this.getStatus(payment_status as StripeTransactionStatus),
         params: {
           checkoutStatus: status as StripeCheckoutStatus,
           paymentStatus: payment_status as StripeTransactionStatus,
         },
       },
-      id,
     );
     /* eslint-enable camelcase */
   }
+
+  /**
+   * Get and calculate transfer information
+   * @protected
+   */
+  protected async getTransferInfo(
+    entityId: string,
+    userId: string,
+  ): Promise<ITransferInfo | undefined> {
+    const transactions = await this.transactionRepository.find({
+      select: ['amount', 'userId'],
+      where: { entityId, status: TransactionStatus.SUCCESS },
+    });
+
+    const destinationUser = await this.customerRepository.findOne({
+      userId,
+    });
+
+    if (!destinationUser?.params.accountId) {
+      Log.error(
+        'Destination user who is being used for transfer doesnt have the connected account id',
+      );
+
+      return;
+    }
+
+    return transactions.reduce(
+      (previousValue, transaction) => ({
+        ...previousValue,
+        amount: previousValue.amount + transaction.amount,
+      }),
+      {
+        amount: 0,
+        destinationUser: destinationUser.params.accountId,
+        userId: transactions[0].userId,
+      },
+    );
+  }
+
+  /**
+   * Create transfer for connected account
+   */
+  public async createTransfer(entityId: string, userId: string) {
+    const { payoutCoeff } = await remoteConfig();
+    const transfer = await this.getTransferInfo(entityId, userId);
+
+    if (!transfer) {
+      Log.error(`There is no actual transfers for entity with following id: ${entityId}`);
+
+      return;
+    }
+
+    const { id } = await this.paymentEntity.transfers.create({
+      amount: transfer.amount * payoutCoeff,
+      currency: 'usd',
+      destination: transfer.destinationUser,
+    });
+
+    const transaction = this.transactionRepository.create({
+      transactionId: id,
+      userId: transfer.userId,
+      entityId,
+      amount: transfer.amount,
+      type: TransactionType.DEBIT,
+      status: TransactionStatus.INITIAL,
+    });
+
+    await this.transactionRepository.save(transaction);
+  }
+
+  /**
+   * Creates payout transfers for given entities
+   */
+  public payout(entitiesId: { id: string; userId: string }[]) {
+    entitiesId.forEach(({ id, userId }) => {
+      void this.createTransfer(id, userId);
+    });
```

Comment:
you need use `for in` and `await Promise.all or allSetted`

Reply:
Can i use map for this? Is my way correct?